pkg/events/events.go

```diff
@@ -1,57 +1,25 @@
 package events
 
 import (
-	"fmt"
-	"time"
-
-	"github.com/filecoin-project/go-state-types/big"
 	"github.com/filecoin-project/lassie/pkg/types"
-	"github.com/ipfs/go-cid"
 	"github.com/libp2p/go-libp2p/core/peer"
 	"github.com/multiformats/go-multicodec"
 )
 
-var (
-	_ types.RetrievalEvent = RetrievalEventCandidatesFound{}
-	_ types.RetrievalEvent = RetrievalEventCandidatesFiltered{}
-	_ types.RetrievalEvent = RetrievalEventConnected{}
-	_ types.RetrievalEvent = RetrievalEventProposed{}
-	_ types.RetrievalEvent = RetrievalEventAccepted{}
-	_ types.RetrievalEvent = RetrievalEventFirstByte{}
-	_ types.RetrievalEvent = RetrievalEventFailed{}
-	_ types.RetrievalEvent = RetrievalEventSuccess{}
-	_ types.RetrievalEvent = RetrievalEventFinished{}
-)
-
-type EventWithCandidates interface {
-	types.RetrievalEvent
-	Candidates() []types.RetrievalCandidate
-}
-
-type baseEvent struct {
-	eventTime      time.Time
-	retrievalId    types.RetrievalID
-	phaseStartTime time.Time
-	payloadCid     cid.Cid
-	protocols      []multicodec.Code
-}
-
-func (r baseEvent) Time() time.Time                { return r.eventTime }
-func (r baseEvent) RetrievalId() types.RetrievalID { return r.retrievalId }
-func (r baseEvent) PhaseStartTime() time.Time      { return r.phaseStartTime }
-func (r baseEvent) PayloadCid() cid.Cid            { return r.payloadCid }
-func (r baseEvent) Protocols() []multicodec.Code   { return r.protocols }
-
-type indexerEvent struct {
-	baseEvent
-	candidates []types.RetrievalCandidate
-}
-
-func (r indexerEvent) StorageProviderId() peer.ID             { return peer.ID("") }
-func (r indexerEvent) Candidates() []types.RetrievalCandidate { return r.candidates }
-
-type RetrievalEventCandidatesFound struct {
-	indexerEvent
+// Identifier returns the peer ID of the storage provider if this retrieval was
+// requested via peer ID, or the string "Bitswap" if this retrieval was
+// requested via the Bitswap protocol
+func Identifier(evt types.RetrievalEvent) string {
```

Comment:
why did this get moved out of types.go? did we not end up needing it anywhere else?